### ISSUE TITLE
FXAA applied multiple times in UI

### ISSUE THREAD

**Issue by fluffrabbit** [godotengine/godot#6578]
![fxaaproblem](https://cloud.githubusercontent.com/assets/6635630/18718432/7c7a20c6-7ff1-11e6-87bc-d489f365e718.png)

In the top image, there are thicker blur parts.

In the bottom image, there are thinner blur parts characteristic of a lightweight FXAA implementation.

The top image is obviously incorrect and represents no known version of FXAA.

The issue seems to be that FXAA is applied to the whole rendering surface rather than just to a 3D camera. It's a very subtle difference, but the current pipeline makes FXAA kind of funny in some cases. What are those cases?

The top image is a screenshot from the editor in which the scene node tree looks like this:

> WorldEnvironment with FXAA
> 
> > Stuff

This problem may also surface >>>ingame<<< in cases like:

> Scene
> 
> > Viewport
> > 
> > > Camera

What is happening? Here is what is happening:

Godot is like, "You want to render a scene, and you want FXAA? Okay, here's FXAA. Now you want that scene in a viewport, and you want FXAA? Okay, here's FXAA."

Bottom line, FXAA should only be applied once.

This issue was hinted at in #654 before but the discussion was less extensive.

**Comment by bojidar-bg** [comment]
Is this valid in 3.0/master as well?

**Comment by fluffrabbit** [comment]
@bojidar-bg Testing is as simple as turning on FXAA and comparing how it looks in the editor to how it looks ingame (assuming your game renders to the screen rather than directly to a nested viewport). Fixing is another story, as solving the problem with a simple automatic on/off flag would leave out corner cases like for instance one's own VR implementation, 3D-in-2D-in-3D, etc. There needs to be some kind of sophisticated AI routine to solve this problem. There is no such thing as over-programming, and Godot is such a comprehensive solution that I have faith in the supergeniusness of the developers.

But anyhow if you want to test please do so. Or someone else could if you don't have a working build environment. As for myself, I'm just a soon-to-be-homeless high-sodium-diet Ubuntu-14-using loser who can't afford the high-speed internet to download Godot's repo or even new Godot builds and I'm no longer welcome in the local Starbucks after I cussed out the staff for not offering me a receipt.

**Comment by fluffrabbit** [comment]
Godot 3.0 builds are broken and will be for a few months. Building for X11 seems to partially work but is spotty and I was not able to test. FXAA is still broken on Godot 2.1.2.

**Comment by Noshyaar** [comment]
First of all thank you for your report and sorry for the delay.

We released Godot 3.0 in January 2018 after 18 months of work, fixing many old issues either directly, or by obsoleting/replacing the features they were referring to.

We still have hundreds of issues whose relevance/reproducibility needs to be checked against the [current stable version](https://godotengine.org/download), and that's where you can help us.
**Could you check if the issue that you described initially is still relevant/reproducible in Godot 3.0 or any newer version, and comment about it here?**

For bug reports, please also make sure that the issue contains detailed steps to reproduce the bug and, if possible, a zipped project that can be used to reproduce it right away. This greatly speeds up debugging and bugfixing tasks for our contributors.

Our Bugsquad will review this issue more in-depth in 15 days, and potentially close it if its relevance could not be confirmed.

Thanks in advance.

*Note: This message is being copy-pasted to many "stale" issues (90+ days without activity). It might happen that it is not meaningful for this specific issue or appears oblivious of the issue's context, if so please comment to notify the Bugsquad about it.*

**Comment by akien-mga** [comment]
As there was no answer to the above request, we're closing this issue as inactive.
If you can still reproduce it, please open a new issue against a currently supported release of Godot.
